client/components/date-range/header.tsx

```diff
@@ -1,47 +1,72 @@
-import { Button } from '@automattic/components';
+import { Gridicon, Button } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
 import { FunctionComponent } from 'react';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const noop = () => {};
-
 interface Props {
-	onApplyClick: () => void;
-	onCancelClick: () => void;
-	applyButtonText: string | null | undefined;
-	cancelButtonText: string | null | undefined;
+	customTitle?: string;
+	startDate: Date | null;
+	endDate: Date | null;
+	resetDates: () => void;
 }
 
 const DateRangeHeader: FunctionComponent< Props > = ( {
-	onCancelClick = noop,
-	onApplyClick = noop,
-	cancelButtonText,
-	applyButtonText,
+	customTitle,
+	startDate,
+	endDate,
+	resetDates,
 } ) => {
 	const translate = useTranslate();
 
-	const cancelText = cancelButtonText || translate( 'Cancel' );
-	const applyText = applyButtonText || translate( 'Apply' );
+	// Add this check at the beginning of the component
+	if ( startDate === undefined || endDate === undefined || resetDates === undefined ) {
+		return null; // or return a loading state
+	}
+
+	const renderDateHelp = () => {
+		return (
+			<div className="date-range__info" role="status" aria-live="polite">
+				{ ! startDate &&
+					! endDate &&
+					translate( '{{icon/}} Please select the {{em}}first{{/em}} day.', {
+						components: {
+							icon: <Gridicon aria-hidden="true" icon="info" />,
+							em: <em />,
+						},
+					} ) }
+				{ startDate &&
+					! endDate &&
+					translate( '{{icon/}} Please select the {{em}}last{{/em}} day.', {
+						components: {
+							icon: <Gridicon aria-hidden="true" icon="info" />,
+							em: <em />,
+						},
+					} ) }
+				{ startDate && endDate && (
+					<Button
+						className="date-range__info-btn"
+						borderless
+						compact
+						onClick={ resetDates }
+						aria-label={ translate( 'Reset selected dates' ) }
+					>
+						{ translate( '{{icon/}} reset selected dates', {
+							components: { icon: <Gridicon aria-hidden="true" icon="cross-small" /> },
+						} ) }
+					</Button>
+				) }
+			</div>
+		);
+	};
 
 	return (
 		<div className="date-range__popover-header">
-			<Button
-				className="date-range__cancel-btn"
-				onClick={ onCancelClick }
-				compact
-				aria-label={ cancelText }
-			>
-				{ cancelText }
-			</Button>
-			<Button
-				className="date-range__apply-btn"
-				onClick={ onApplyClick }
-				primary
-				compact
-				aria-label={ applyText }
-			>
-				{ applyText }
-			</Button>
+			<div className="date-range__controls">
+				{ customTitle ? (
```

Comment:
This is definitely isn't a blocker, but I'm still thinking maybe we shouldn't hide the help functions when there is a custom title, i.e. it'll be better if the custom title and the helpers could co-exist.

Just something for you to consider. Not a priority right now.

Great work!

Reply:
Thanks, I agree there may be some way for them to work together in a future iteration. We will figure something out!